src/main/java/dev/jbang/dependencies/ArtifactResolver.java

```diff
@@ -182,16 +182,19 @@ public List<ArtifactInfo> resolve(List<String> depIds) {
 															.collect(Collectors.groupingBy(Dependency::getScope));
 
 			List<Dependency> deps = scopeDeps.get(JavaScopes.COMPILE);
-			List<Dependency> managedDeps = null;
+			List<Dependency> managedDeps = deps	.stream()
+												.flatMap(d -> getManagedDependencies(d).stream())
+												.collect(Collectors.toList());
+
 			if (scopeDeps.containsKey("import")) {
 				// If there are any @pom artifacts we'll apply their
-				// managed dependencies to the given dependencies
+				// managed dependencies to the given dependencies BEFORE ordinary deps
 				List<Dependency> boms = scopeDeps.get("import");
 				List<Dependency> mdeps = boms	.stream()
 												.flatMap(d -> getManagedDependencies(d).stream())
 												.collect(Collectors.toList());
 				deps = deps.stream().map(d -> applyManagedDependencies(d, mdeps)).collect(Collectors.toList());
-				managedDeps = mdeps;
+				managedDeps.addAll(0, mdeps);
```

Comment:
if we do this do we even need to set dependencies?

Reply:
oh we do because managed dependences aren't resolved, just used for guiding proper resolution.

Reply:
actually - why does it fix things to add all deps to managed dependencies?

Reply:
No, we don't do that. What we do is "pull all dependency dependencyManagement section (if any)." Dependencies are still on level 1, it is only their management section pulled up.